format/avro/codecs/enum.go

```diff
@@ -0,0 +1,24 @@
+package codecs
+
+import (
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/pkg/decode"
+)
+
+type EnumCodec struct {
+	symbols []string
+}
+
+func (l EnumCodec) Decode(name string, d *decode.D) {
+	d.FieldScalarStrFn(name, func(d *decode.D) string {
```

Comment:
It would be nice use `scalar.Mapper` here then all enums would have both their "raw" (called actual atm) or their symbolic value accessible.

Something like this:

```go
func (l EnumCodec) MapScalar(s scalar.S) (S, error) {
	// some casting and bounds checks needed here
	// ActualS returns the decoded zigzag signed int64
	if t, ok := l.symbols[s.ActualS()]; ok {
		s.Sym = t
	}
	return s, nil
}

func (l EnumCodec) Decode(name string, d *decode.D) {
	d.FieldSFn(name, VarZigZag, l)
}
```

Reply:
Good idea, updated